cirq/contrib/two_qubit_gates/gate_compilation.py

```diff
@@ -0,0 +1,328 @@
+                                mesh_points, (u_locals_0, u_locals_1),
+                                (u_locals_0, u_locals_1))
+
+    kak_vecs.extend(out[0])
+    sq_cycles.extend(out[1])
+
+    if verbose:
+        # coverage: ignore
+        print(f'fraction satisfied with 2 gates and 3 gates(same single qubit)'
+              f': {len(u_locals_for_gate) / mesh_points.shape[0]:.3f}')
+
+    # If all KAK vectors in the mesh have been tabulated, return.
+    missing_vec_inds = set(range(mesh_points.shape[0])) - set(u_locals_for_gate)
+
+    if not missing_vec_inds:
+        # coverage: ignore
+        return GateTabulation(base_gate, np.array(kak_vecs), sq_cycles,
+                              max_infidelity)
+
+    # Run through remaining KAK vectors that don't have products and try to
+    # correct them
+
+    u_locals_0p = random_qubit_unitary(100)
+    u_locals_1p = random_qubit_unitary(100)
+    u_locals = vector_kron(u_locals_0p, u_locals_1p)
+
+    # Loop through the mesh points that have not yet been tabulated.
+    # Consider their nonlocal parts A and compute products of the form
+    # base_gate^\dagger k A
+    # Compare the KAK vector of any of those products to the already tabulated
+    # KAK vectors from single products of the form
+    # base_gate k0 base_gate.
+    # If they are close, then base_gate^\dagger k A  ~ base_gate k0 base_gate
+    # So we may compute the outer local unitaries kL, kR such that
+    #    base_gate^\dagger k A = kL base_gate k0 base_gate kR
+    #    A = k^\dagger base_gate kL base_gate k0 base_gate kR
+    #    the single-qubit unitary kL is the one we need to get the desired
+    #    KAK vector.
+    base_gate_dag = base_gate.conj().T
+    for ind in missing_vec_inds:
+        missing_vec = mesh_points[ind]
+        missing_unitary = KAK_vector_to_unitary(missing_vec)
+
+        products = np.einsum('ab,...bc,cd', base_gate_dag, u_locals,
+                             missing_unitary)
+        kaks = kak_vector(products, check_preconditions=False)
+        kaks = kaks[..., np.newaxis, :]
+
+        dists2 = np.sum((kaks - kak_vecs_single)**2, axis=-1)
+        min_dist_inds = np.unravel_index(dists2.argmin(), dists2.shape)
+        min_dist = np.sqrt(dists2[min_dist_inds])
+        if min_dist < tabulation_cutoff:
+            new_ind, old_ind = min_dist_inds
+
+            old_sq_cycle = sq_cycles_single[old_ind][0]
+            old_k = np.kron(*old_sq_cycle)
+            base_product = base_gate @ old_k @ base_gate
+            new_product = products[new_ind]
+
+            kRkL, actual = _outer_locals_for_unitary(new_product, base_product)
+            # Add to the enumeration
+            sq_cycles.append((old_sq_cycle, kRkL[1]))
+            kak_vecs.append(
+                kak_vector(base_gate @ actual, check_preconditions=False))
+        elif include_warnings:
+            # coverage: ignore
+            warn(f'Failed to tabulate a KAK vector near {missing_vec}')
```

Comment:
If the user wants all points to be covered, this should be an error instead of a warning.

Add a method to `GateTabulation` to print any missed vectors. Change `include_warnings` to `allow_missed_points`. If a point is missed and `allow_missed_points` is false, raise an error.

Reply:
Done.